tests/post/steps/test_ingress.py

```diff
@@ -125,6 +204,37 @@ def check_cp_ingress_node_ip(control_plane_ingress_ip, node_name, ssh_config):
     assert control_plane_ingress_ip == ip
 
 
+@then(parsers.parse("the control plane ingress IP is equal to '{ip}'"))
+def check_cp_ingress_ip(control_plane_ingress_ip, ip):
+    assert control_plane_ingress_ip == ip
+
+
+def get_node_hosting_cp_ingress_vip(k8s_client):
+    # To get the node where sit the VIP we need to look at event on the Service
+    field_selectors = [
+        "reason=nodeAssigned",
+        "involvedObject.kind=Service",
+        "involvedObject.name=ingress-nginx-control-plane-controller",
+    ]
+    events = k8s_client.list_namespaced_event(
+        "metalk8s-ingress",
+        field_selector=",".join(field_selectors),
+    )
+
+    assert events.items, "Unable to get event for Control Plane Ingress Service"
+
+    match = re.search(
+        r'announcing from node "(?P<node>.+)"',
+        sorted(events.items, key=lambda event: event.last_timestamp, reverse=True)[
+            0
+        ].message,
+    )
```

Comment:
Is this robust enough? I'm wondering if we may get other events matching the field selectors that won't match this regular expression.

Reply:
Since we filter on this specific service with only "reason=nodeAssigned" to me it's safe and I do not see any better approach for this (already tried relying directly on ARP but it's a mess since sometimes you may run these tests on the machine hosting the VIP currently, so you do not have the mac address since it's not resolved locally in arp table ... )

Reply:
> Since we filter on this specific service with only "reason=nodeAssigned" to me it's safe

OK :ok_hand: 

I guess we could have been slightly more conservative with e.g.

```suggestion
    match = None
    for event in sorted(events.items, key=lambda event: event.last_timestamp, reverse=True):
        match = re.search(r'announcing from node "(?P<node>.+)"', event.message)
        if match is not None:
            break
```

Reply:
Right, we can do this also, not sure it brings any value today but ... it does not cost much so I will do